cmd/cmd.go

```diff
@@ -529,11 +529,27 @@ func generate(cmd *cobra.Command, model, prompt string, wordWrap bool, format st
 	}
 
 	if err := client.Generate(cancelCtx, &request, fn); err != nil {
-		if strings.Contains(err.Error(), "context canceled") && abort {
+		switch {
+		case strings.Contains(err.Error(), "context canceled") && abort:
 			return nil
+		case strings.Contains(err.Error(), "unsupported model format"):
+			// pull and retry to see if the model has been updated
+			if err := PullHandler(cmd, []string{model}); err != nil {
+				fmt.Printf("Error: %s\n", err)
+				return fmt.Errorf("unsupported, please update this model to gguf format") // relay the original error
```

Comment:
This model is no longer supported by Ollama. A new version of this model may be available: ollama pull <model name>

Reply:
We should also print a different message based on `library/` vs other models, since those are most likely built locally.

Reply:
Lastly, for `library/` models, let's instead print out a message and auto `ollama pull`

```
% ollama run llama2
This model is no longer compatible with Ollama. Pulling a new version...
<pulls>
```